middleware/authz.js

```diff
@@ -0,0 +1,22 @@
+require("express-async-errors");
+const jwt = require("jsonwebtoken");
+
+module.exports =  (listOfRolesAllowed) => {
+  return async (req, res, next) => {
+    const token = req.header("x-auth-token");
+    if (!token)
+      return res.status(401).send(`Access denied. No token provided.`);
```

Comment:
Please coordinate with your teammates and review any comments I’ve provided on their work. I’ve encountered similar code issues previously and have already suggested improvements that could be relevant to this context.